nixos/tests/ente/default.nix

```
@@ -0,0 +1,139 @@
+      };
+    };
+
+  nodes.ente =
+    {
+      config,
+      nodes,
+      lib,
+      ...
+    }:
+    {
+      security.pki.certificateFiles = [ certs.ca.cert ];
+
+      networking.extraHosts = ''
+        ${config.networking.primaryIPAddress} accounts.${domain} albums.${domain} api.${domain} cast.${domain} photos.${domain} s3.${domain}
+      '';
+
+      networking.firewall.allowedTCPPorts = [
+        80
+        443
+      ];
+
+      services.nginx = {
+        recommendedProxySettings = true;
+        virtualHosts =
+          lib.genAttrs
+            [
+              "accounts.${domain}"
+              "albums.${domain}"
+              "api.${domain}"
+              "cast.${domain}"
+              "photos.${domain}"
+            ]
+            (_: {
+              sslCertificate = certs.${domain}.cert;
+              sslCertificateKey = certs.${domain}.key;
+            })
+          // {
+            "s3.${domain}" = {
+              forceSSL = true;
+              sslCertificate = certs.${domain}.cert;
+              sslCertificateKey = certs.${domain}.key;
+              locations."/".proxyPass = "http://${nodes.minio.networking.primaryIPAddress}:9000";
+              extraConfig = ''
+                client_max_body_size 32M;
+                proxy_buffering off;
+                proxy_request_buffering off;
+              '';
+            };
+          };
+      };
+
+      services.ente = {
+        web = {
+          enable = true;
+          domains = {
+            accounts = "accounts.${domain}";
+            albums = "albums.${domain}";
+            cast = "cast.${domain}";
+            photos = "photos.${domain}";
+          };
+        };
+        api = {
+          enable = true;
+          nginx.enable = true;
+          enableLocalDB = true;
+          domain = "api.${domain}";
+          settings = {
+            s3 = {
+              use_path_style_urls = true;
+              b2-eu-cen = {
+                endpoint = "https://s3.${domain}";
+                region = "us-east-1";
+                bucket = "ente";
+                key._secret = pkgs.writeText "accesskey" accessKey;
+                secret._secret = pkgs.writeText "secretkey" secretKey;
+              };
+            };
+            key = {
+              encryption._secret = pkgs.writeText "encryption" "T0sn+zUVFOApdX4jJL4op6BtqqAfyQLH95fu8ASWfno=";
+              hash._secret = pkgs.writeText "hash" "g/dBZBs1zi9SXQ0EKr4RCt1TGr7ZCKkgrpjyjrQEKovWPu5/ce8dYM6YvMIPL23MMZToVuuG+Z6SGxxTbxg5NQ==";
+            };
+            jwt.secret._secret = pkgs.writeText "jwt" "i2DecQmfGreG6q1vBj5tCokhlN41gcfS2cjOs9Po-u8=";
+          };
+        };
+      };
+    };
+
+  testScript = ''
+    minio.start()
+    minio.wait_for_unit("minio.service")
+    minio.wait_for_open_port(9000)
+
+    # Create a test bucket on the server
+    minio.succeed("mc config host add minio http://localhost:9000 ${accessKey} ${secretKey} --api s3v4")
```

Test is failing here, I think the `config` subcommand has been deprecated: https://github.com/minio/mc/issues/5206

```suggestion
    minio.succeed("mc alias set minio http://localhost:9000 ${accessKey} ${secretKey} --api s3v4")
```

---

Good catch! It has indeed been deprecated: https://help.ente.io/self-hosting/troubleshooting/docker#minio-provisioning-error

---

Yep good catch, thanks!